collections/triple.go

```diff
@@ -274,30 +274,57 @@ func (t tripleKeyCodec[K1, K2, K3]) SizeNonTerminal(key Triple[K1, K2, K3]) int
 }
 
 // NewPrefixUntilTripleRange defines a collection query which ranges until the provided Pair prefix.
-// Unstable: this API might change in the future.
-func NewPrefixUntilTripleRange[K1, K2, K3 any](k1 K1) Ranger[Triple[K1, K2, K3]] {
+// Unstable: this API might change in the future. Use the first boolean option to get a reverse iterator (reverse=true).
+func NewPrefixUntilTripleRange[K1, K2, K3 any](k1 K1, opts ...bool) Ranger[Triple[K1, K2, K3]] {
```

Comment:
No, let's not have a variadic of boolean, it doesn't have a good UX.
If we go for variadic, let's use an Option struct and make those take options: this pattern: https://github.com/cosmos/cosmos-sdk/blob/main/log/options.go#L19-L47

Reply:
technically also variadic is breaking, eg if a function was typechecked etc.

I'd prefer we have a separate function, cc @oren-lava

Reply:
done. see https://github.com/oren-lava/cosmos-sdk/commit/3cf23b942b76f7d99d0c7c58f979c80045c4320d

Reply:
done. see https://github.com/cosmos/cosmos-sdk/pull/21496/commits/bc0f0607bb11b8e1588430bd38041f183061626b

Reply:
reverted the variadic change and wrote separate functions. see https://github.com/cosmos/cosmos-sdk/pull/21496/commits/7e970f2834a5957c6138f6ed80e75d8cb4ed97d1

cc @testinginprod , @julienrbrt